pkg/kubecost/allocation.go

```diff
@@ -1539,7 +1542,56 @@ func (as *AllocationSet) AggregateBy(aggregateBy []string, options *AllocationAg
 					if alloc.SharedCostBreakdown == nil {
 						alloc.SharedCostBreakdown = map[string]SharedCostBreakdown{}
 					}
-					sharedCostName := sharedAlloc.generateKey(aggregateBy, options.LabelConfig)
+
+					var sharedAggregateBy []string
+					for labelKey, _ := range options.SharedLabels {
+						sharedAgg := fmt.Sprintf("label:%s", labelKey)
+						if !slices.Contains(sharedAggregateBy, sharedAgg) {
+							sharedAggregateBy = append(sharedAggregateBy, sharedAgg)
+						}
+					}
+					if len(options.SharedNamespaces) > 0 {
+						sharedAggregateBy = append(sharedAggregateBy, "namespace")
+					}
+					sharedCostName := sharedAlloc.generateKey(sharedAggregateBy, options.LabelConfig)
+
+					// get each value in the generated key, then reset the name
+					sharedCostNameValues := strings.Split(sharedCostName, "/")
+					sharedCostName = ""
+
+					// try to match to the first label
+					for _, sharedLabelValues := range options.SharedLabels {
+						for _, sharedCostLabel := range sharedCostNameValues {
```

Comment:
I don't think you exactly need (or want) to check each label value with each name because then you'll end up comparing label values with namespace values (as well as label values with unrelated label values).

e.g. if you are sharing by `app:cost-analyzer, env:test, namespace:kubecost` and you get the key `__unallocated__/test/default` you just want to compare:
1. `__unallocated__ == cost-analyzer` ? false
2. `test == test` ? true

This code would do something slightly different:
1. `cost-analyzer == __unallocated__` ? false
2. `cost-analyzer == test` ? false
3. `test == __unallocated__` ? false
4. `test == test` ? true

Not only is this more inefficient, it's prone to accidental matches, where you're comparing a label to a value the wrong value and might _accidentally match_ on it.

I think we just need a more linear checking of (first label matches first value?), (second label matches second value?) ... in that order. Does that make sense?